pyproject.toml

```diff
@@ -20,3 +20,80 @@ multi_line_output = 3
 include_trailing_comma = "True"
 force_grid_wrap = 0
 use_parentheses = "True"
+
+[project]
+name = "mozregression"
+dynamic = ["version"]
+description = "Regression range finder for Mozilla nightly builds"
+readme = "README.md"
+dependencies = [
+  "glean_sdk>=60.3.0",
+  "beautifulsoup4>=4.7.1",
+  "colorama>=0.4.1",
+  "configobj>=5.0.6",
+  "distro>=1.8.0",
+  "importlib_resources>= 5.10",
+  "mozdevice>=4.1.0,<5",
+  "mozfile>=2.0.0",
+  "mozinfo>=1.1.0",
+  "mozinstall>=2.0.0",
+  "mozlog>=4.0",
+  "mozprocess>=1.3.1",
+  "mozprofile>=2.2.0",
+  "mozrunner>=8.0.2",
+  "mozversion>=2.1.0",
+  "redo>=2.0.2",
+  "requests>=2.21.0",
+  "taskcluster>=6.0.0",
+]
+requires-python = ">= 3.8.1"
+license = {text = "MPL 2.0"}
+authors = [
+  { name = "Mozilla Automation and Tools Team", email = "tools@lists.mozilla.org" }
+]
+classifiers=[
+  "License :: OSI Approved :: Mozilla Public License 2.0 (MPL 2.0)",
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Console",
+  "Intended Audience :: Developers",
+  "Operating System :: OS Independent",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+  "Programming Language :: Python :: 3 :: Only",
+]
+
+[project.scripts]
+mozregression = "mozregression.main:main"
+
+[project.optional-dependencies]
+dev = [
+  "coverage==7.6.1",
+  "coveralls==4.0.1",
+  "mock==5.1.0",
+  "pytest==8.3.2",
+  "pytest-mock==3.14.0",
+]
+gui = [
+  "PySide6==6.6.*",
+]
+gui-dev = [
+  "pytest-qt==4.4.0",
+]
+linters = [
+  "flake8>=6.0.0",
+  "flake8-black",
+  "glean-parser",
+  "isort",
+  "mccabe",
+  "pyflakes",
+]
```

Comment:
I think for the most part, this section of the code will not actually be used and adding it might make it harder to maintain these lists. E.g., the "gui" is built via pyinstaller, and distributed via GitHub in CI (based on the requirements files), and the dev/linters is also installed the same way. What do you think?

Reply:
That depends on what package manager is used, but given the recommended installation is to use the requirements.txt, I think we can skip this here.